test/functional/tests/lazy_writes/test_lazy_writes_clean.py

```diff
@@ -0,0 +1,259 @@
+        if test_file_md5sum_main != test_file_3.md5sum():
+            TestRun.LOGGER.error("Md5 sums should be equal.")
+
+    with TestRun.step("Delete test files."):
+        test_file_main.remove(True)
+        test_file_1.remove(True)
+        test_file_2.remove(True)
+        test_file_3.remove(True)
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode.with_traits(CacheModeTrait.LazyWrites))
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_clean_remove_core(cache_mode):
+    """
+        title: Test of the ability to remove core from cache in modes with lazy writes.
+        description: |
+          Test if OpenCAS removes core in modes with lazy writes without data loss.
+        pass_criteria:
+          - Core removing works properly.
+          - Data on core device is correct after core is removed.
+    """
+    with TestRun.step("Prepare devices for cache and core."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(256, Unit.MebiByte)])
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(512, Unit.MebiByte)])
+        core_part = core_dev.partitions[0]
+        Udev.disable()
+
+    with TestRun.step(f"Start cache in {cache_mode} mode."):
+        cache = casadm.start_cache(cache_part, cache_mode)
+
+    with TestRun.step("Add core to cache."):
+        core_part.create_filesystem(Filesystem.xfs)
+        core = cache.add_core(core_part)
+        core.mount(mnt_point)
+
+    with TestRun.step("Disable cleaning and sequential cutoff policies."):
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+        # get IO stats independent from test
+        exp_obj_stats_diff = IOstat.get_iostat(core).total_writes.value
+        core_disk_writes_diff = IOstat.get_iostat(core_part).total_writes.value
+
+    with TestRun.step("Write data to exported object."):
+        test_file_main = create_random_test_file("/tmp/test_file_main", Size(64, Unit.MebiByte))
+        dd = Dd().output(mnt_point + "/test_file") \
+            .input(test_file_main.full_path) \
+            .block_size(bs) \
+            .count(int(test_file_main.size / bs)) \
+            .oflag("direct")
+        dd.run()
+        test_file_main.refresh_item()
+        test_file_md5sum_main = test_file_main.md5sum()
+
+    with TestRun.step("Check IO stats."):
+        core_disk_writes = IOstat.get_iostat(core_part).total_writes.value - core_disk_writes_diff
+        exp_obj_stats = IOstat.get_iostat(core).total_writes.value - exp_obj_stats_diff
+        if core_disk_writes > 0:
+            TestRun.LOGGER.error("Writes should occur only on cache device.")
+        if exp_obj_stats != test_file_main.size.value:
+            TestRun.LOGGER.error("Not all writes reached exported object.")
```

Comment:
You have a FS on core device. Please run this test for each FS and check if there is no inconsistency in stats caused by filesystem metadata writes. You can workaround it by adding ioclass for metadata with disabled selective allocation.